src/mrt/nodejs_npm/info.rs

```diff
@@ -0,0 +1,57 @@
+use std::{path::PathBuf, fs::File, io::BufReader};
+use anyhow::{Result, Context};
+use serde::Deserialize;
+
+use crate::{package::PackageInfoExtractor};
+
+pub struct NpmPackageInfoExtractor {
+    package_json: NpmPackageJSON
+}
+
+#[derive(Deserialize, Debug)]
+struct NpmPackageJSON {
+    name: String,
+    version: String,
+}
+
+impl NpmPackageInfoExtractor {
+    pub fn from_package_path(package_path: &PathBuf) -> Result<NpmPackageInfoExtractor> {
+
+        let package_json_path = package_path.join("package.json");
+
+        let package_json_file = File::open(&package_json_path)
+            .context(format!("Reading package.json at {}", &package_json_path.display()))?;
+        let package_json_reader = BufReader::new(package_json_file);
+
+        let package_json = serde_json::from_reader(package_json_reader)
+            .context(format!("Parsing JSON of package.json at {}", package_json_path.display()))?;
+
+        return Ok(NpmPackageInfoExtractor {
+            package_json
+        });
+    }
+}
+
+impl PackageInfoExtractor for NpmPackageInfoExtractor {
+    fn get_name(&self) -> &str {
+        return self.package_json.name.as_str();
+    }
+
+    fn get_version(&self) -> &str {
+        return self.package_json.version.as_str();
+    }
+}
+
+#[test]
+fn from_package_path_no_folder_exists() {
+    let non_existing_package_path = PathBuf::from("does-not-exist");
+    let result = NpmPackageInfoExtractor::from_package_path(&non_existing_package_path);
+    assert!(result.is_err());
+}
+
+#[test]
+fn from_package_path_success() {
```

Comment:
what happens if `package.json` have broken content. For example with merge markers inside? That maybe feature request, but what if content is completely bogus, what do you expect from your lib?

Reply:
It will result error, I added improvedd error message.